src/surface.c

```diff
@@ -0,0 +1,61 @@
+#include "surface.h"
+#include "n64sys.h"
+#include "debug.h"
+#include <assert.h>
+#include <string.h>
+
+const char* tex_format_name(tex_format_t fmt)
+{
+    switch (fmt) {
+    case FMT_NONE:   return "FMT_NONE";
+    case FMT_RGBA16: return "FMT_RGBA16";
+    case FMT_YUV16:  return "FMT_YUV16";
+    case FMT_CI4:    return "FMT_CI4";
+    case FMT_CI8:    return "FMT_CI8";
+    case FMT_IA4:    return "FMT_IA4";
+    case FMT_IA8:    return "FMT_IA8";
+    case FMT_IA16:   return "FMT_IA16";
+    case FMT_I4:     return "FMT_I4";
+    case FMT_I8:     return "FMT_I8";
+    default:         return "FMT_???";
+    }
+}
+
+surface_t surface_alloc(tex_format_t format, uint32_t width, uint32_t height)
+{
+    return (surface_t){ 
+        .flags = format | SURFACE_FLAGS_OWNEDBUFFER,
+        .width = width,
+        .height = height,
+        .stride = TEX_FORMAT_PIX2BYTES(format, width),
+        .buffer = malloc_uncached_aligned(64, height * TEX_FORMAT_PIX2BYTES(format, width)),
```

Comment:
See my comment in surface.h

Reply:
This should be fixed as well with the roundup.